server/graphql/v2/query/collection/OrdersCollectionQuery.ts

```diff
@@ -314,12 +323,18 @@ export const OrdersCollectionResolver = async (args, req: express.Request) => {
   if (args.expectedDateFrom) {
     where['data.expectedAt'] = { [Op.gte]: args.expectedDateFrom };
   }
-
   if (args.expectedDateTo) {
     where['data.expectedAt'] = where['data.expectedAt'] || {};
     where['data.expectedAt'][Op.lte] = args.expectedDateTo;
   }
 
+  if (args.chargedDateFrom) {
+    where[Op.and].push({ ['$Subscription.lastChargedAt$']: { [Op.gte]: args.chargedDateFrom } });
+  }
+  if (args.chargedDateTo) {
+    where[Op.and].push({ ['$Subscription.lastChargedAt$']: { [Op.lte]: args.chargedDateTo } });
+  }
```

Comment:
The filter is broken for one-time contributions, perhaps a similar solution as for the ordering is needed:

```suggestion
  if (args.chargedDateFrom) {
    where[Op.and].push(
      sequelize.where(sequelize.literal(`COALESCE("Subscription"."lastChargedAt", "Order"."createdAt")`), {
        [Op.gte]: args.chargedDateFrom,
      }),
    );
  }
  if (args.chargedDateTo) {
    where[Op.and].push(
      sequelize.where(sequelize.literal(`COALESCE("Subscription"."lastChargedAt", "Order"."createdAt")`), {
        [Op.lte]: args.chargedDateTo,
      }),
    );
  }
```